src/Model_Hydro/CPU_Hydro/CPU_FluidSolver_MHM.cpp

```diff
@@ -341,11 +342,13 @@ void CPU_FluidSolver_MHM(
 #     ifdef __CUDACC__
       const int P = blockIdx.x;
 #     else
-#     pragma omp for schedule( runtime ) private ( Iteration, AdaptiveMinModCoeff, State )
+#     pragma omp for schedule( runtime ) private ( Iteration, AdaptiveMinModCoeff, FullStepFailure )
       for (int P=0; P<NPatchGroup; P++)
 #     endif
       {
 #        ifndef __CUDACC__
+         // the private variables FullStepFailure must be initialized within the OpenMP parallel region
+         FullStepFailure = 0;
          Iteration = 0;
 #        endif
```

Comment:
@hyschive We should keep these lines as both `FullStepFailure` and `Iteration` should be re-initialize to zero when looping over all patch groups with a single thread. How do you think?

Reply:
@zengbs You are correct! In that case, I would suggest removing `#  ifndef __CUDACC__` and NOT assigning initial values when declaring these variables (around line 300). Also, use `private` instead of `firstprivate`.

Reply:
This works properly! Thanks!